EIPS/eip-1559.md

```diff
@@ -155,17 +155,24 @@ ELASTICITY_MULTIPLIER = 2
 
 class World(ABC):
 	def validate_block(self, block: Block) -> None:
+		parent_gas_target = self.parent(block).gas_limit // ELASTICITY_MULTIPLIER
+
+		# On the fork block, don't account for the ELASTICITY_MULTIPLIER to avoid
+		# unduly halving the gas target.
+		if INITIAL_FORK_BLOCK_NUMBER == block.number:
+			parent_gas_target = self.parent(block).gas_limit 
+
 		parent_base_fee_per_gas = self.parent(block).base_fee_per_gas
 		parent_gas_used = self.parent(block).gas_used
-		parent_gas_target = self.parent(block).gas_target
+		gas_target = block.gas_limit // ELASTICITY_MULTIPLIER
 		transactions = self.transactions(block)
 
 		# check if the block used too much gas
-		assert block.gas_used <= block.gas_target * ELASTICITY_MULTIPLIER, 'invalid block: too much gas used'
+		assert block.gas_used <= block.gas_limit, 'invalid block: too much gas used'
 
 		# check if the block changed the gas target too much
-		assert block.gas_target <= parent_gas_target + parent_gas_target // 1024, 'invalid block: gas target increased too much'
-		assert block.gas_target >= parent_gas_target - parent_gas_target // 1024, 'invalid block: gas target decreased too much'
+		assert gas_target <= parent_gas_target + parent_gas_target // 1024, 'invalid block: gas target increased too much'
+		assert gas_target >= parent_gas_target - parent_gas_target // 1024, 'invalid block: gas target decreased too much'
```

Comment:
Assuming we want no change in the miner gas targeting formula, I think these need to be 2048 now, since we are tweaking the gas limit rather than the gas target.

Reply:
I don't think so. The miner is still controlling the gas target the same way as before, it just isn't included explicitly in the header.

Reply:
Oops, I just realized that this is targeting `gasTarget` not `gasLimit`.